docs/guides/federation.md

```diff
@@ -2,6 +2,175 @@
 title: Federation
 ---
 
-# Federation
+# Apollo Federation Guide
 
-Documentation coming soon
+Strawberry supports [Apollo Federation][1] out of the box, that means that you
+can create services using Strawberry and federate them via Apollo Gateway.
+
+> _NOTE_: we don’t have a gateway server, you’d need to always use the Apollo
+> Gateway for this.
+
+Apollo Federation allows to combine multiple GraphQL APIs into one. This can be
+extremely useful when working with micro-services.
+
+## Federated schema example
+
+Let’s look at an example on how to implement Apollo Federation using Strawberry,
+We’ll have an application with two federated services.
+
+1. book: a service with all the books we have
+2. reviews: a service with book reviews.
+
+### Books service
+
+```python
+@strawberry.federation.type(keys=["id"])
+class Book:
+    id: strawberry.ID
+    title: str
+
+def get_all_books() -> List[Book]:
+    return [Book(id=1, title="The Dark Tower")]
+
+@strawberry.type
+class Query:
+    all_books: List[Book] = strawberry.field(resolver=get_all_books)
+
+schema = strawberry.federation.Schema(query=Query)
+```
+
+We defined two types, `Book` and `Query`, `Query` has only one field that allows
+to fetch all the books.
+
+Meanwhile `Book` used the `strawberry.federation.type` decorator, as opposed to
+the normal `strawberry.type`, this new decorator extend the base one and allows
+to define federation specific attributes to the type.
+
+In this case we are telling federation that the key to uniquely identify a book
+is the `id` field.
+
+> Federation keys can be thought primary keys. They are used by the gateway to
+> query types between multiple services and then joining them into the augmented
+> type.
+
+### Reviews service
+
+Let’s look at how our review service looks like, we want to define a type for a
+review but also extend the book to have a list of review.
+
+```python
+@strawberry.type
+class Review:
+    body: str
+
+def get_reviews() -> List[Review]:
```

Comment:
I think this function should do something with the current book id to make it clearer that it's dependant on the current book.